packages/react/test/specs/commonTests/isConformant.tsx

```diff
@@ -53,40 +58,40 @@ export default (Component, options: Conformant = {}) => {
 
   const componentType = typeof Component
 
-  // This is added because the component is mounted
+  const helperComponentNames = [
+    ...[ThemeProvider, FelaTheme, Ref, RefFindNode],
+    ...(usesWrapperSlot && [Box]),
+  ].map(getDisplayName)
+
+  const toNextNonTrivialChild = (from: ReactWrapper) => {
+    let current = from.childAt(0)
+
+    if (!current) return current
+
+    while (helperComponentNames.indexOf(current.name()) >= 0) {
+      current = current.childAt(0)
+      if (!current) return current
+    }
+
+    return current
+  }
+
   const getComponent = (wrapper: ReactWrapper) => {
-    // FelaTheme wrapper and the component itself:
-    let component = wrapper
-
-    /**
-     * The wrapper is mounted with Provider, so in total there are three HOC components
-     * that we want to get rid of: ThemeProvider, the actual Component and FelaTheme,
-     * in order to be able to get to the actual rendered result of the component we are testing
-     */
-    _.times(nestingLevel + 3, () => {
-      component = component.childAt(0)
-    })
+    const componentElement = toNextNonTrivialChild(wrapper)
+    let topLevelChildElement = toNextNonTrivialChild(componentElement)
 
-    if (component.type() === FocusZone) {
+    // passing through Focus Zone wrappers
+    if (topLevelChildElement.type() === FocusZone) {
       // another HOC component is added: FocuZone
-      component = component.childAt(0) // skip through <FocusZone>
-      if (component.prop(FOCUSZONE_WRAP_ATTRIBUTE)) {
-        component = component.childAt(0) // skip the additional wrap <div> of the FocusZone
+      topLevelChildElement = topLevelChildElement.childAt(0) // skip through <FocusZone>
```

Comment:
Is there a reason why FocusZone is not in the `helperComponentsNames`?

Reply:
yes - just because the logic itself is a bit different for the focus zone. Wanted to encapsulate the most trivial logic of skipping `service` components - for focus zone we have different behavior, however. And while we might introduce something like `skippedProps` to traversal method's config, it will then obfuscate the component type for which this config is necessary (like, these checks are necessary for FocusZone-related components only - note that currently this 'FocusZone-related' is defined by only _when_ method is called, and not by the type/name of the component)